src/uct/sm/scopy/knem/knem_md.c

```diff
@@ -204,6 +184,38 @@ static ucs_status_t uct_knem_mem_dereg(uct_md_h md,
     return status;
 }
 
+int uct_knem_md_check_mem_reg(uct_md_h md)
+{
+    uint8_t buff;
+    uct_knem_key_t key;
+
+    return (uct_knem_mem_reg_internal(md, &buff, sizeof(buff), 0, 1, &key) ==
+            UCS_OK) &&
+           (uct_knem_mem_dereg_internal(md, &key) == UCS_OK);
+}
+
+ucs_status_t uct_knem_md_query(uct_md_h uct_md, uct_md_attr_v2_t *md_attr)
+{
+    uct_knem_md_t *md = ucs_derived_of(uct_md, uct_knem_md_t);
+
+    md_attr->rkey_packed_size = sizeof(uct_knem_key_t);
+    md_attr->flags            = UCT_MD_FLAG_REG | UCT_MD_FLAG_NEED_RKEY;
+    md_attr->reg_mem_types    = uct_knem_md_check_mem_reg(uct_md) ?
```

Comment:
we also need to remove `UCT_MD_FLAG_REG` flag if can't register

Reply:
Fixed.